src/plans/details/tabs/VirtualMachines/components/DeleteVirtualMachines/DeleteVirtualMachinesButton.tsx

```diff
@@ -0,0 +1,52 @@
+import { type FC, useMemo } from 'react';
+import { isPlanArchived } from 'src/modules/Plans/utils/helpers/getPlanPhase';
+import { useModal } from 'src/modules/Providers/modals/ModalHOC/ModalHOC';
+import { useForkliftTranslation } from 'src/utils/i18n';
+
+import { ToolbarItem } from '@patternfly/react-core';
+
+import VMsActionButton from '../VMsActionButton';
+
+import type { DeleteVirtualMachineProps } from './utils/types';
+import DeleteVirtualMachinesModal from './DeleteVirtualMachinesModal';
+
+const DeleteVirtualMachinesButton: FC<DeleteVirtualMachineProps> = ({ plan, selectedIds }) => {
+  const { t } = useForkliftTranslation();
+  const { showModal } = useModal();
+
+  const onClick = () => {
+    showModal(<DeleteVirtualMachinesModal plan={plan} selectedIds={selectedIds} />);
+  };
+
+  const remainingVms = useMemo(() => {
```

Comment:
this only used inside a reason, no need for additional useMemo , just add it to the useMemo below

Reply:
put it above line 35, so if you already returning before , no need to calc